src/app/inventory/note-hashtags.ts

```diff
@@ -1,26 +1,50 @@
-import { compact, filterMap, uniqBy } from 'app/utils/collections';
+import { compact, filterMap } from 'app/utils/collections';
+import { compareBy } from 'app/utils/comparators';
+import { maxBy } from 'es-toolkit';
 import { ItemInfos } from './dim-item-info';
 
 /**
- * collects all hashtags from item notes
+ * Collects all hashtags from all item notes.
+ *
+ * Orders by use count, de-dupes case-insensitive, and picks the most popular capitalization.
  */
-export function collectNotesHashtags(itemInfos: ItemInfos) {
-  const hashTags = new Set<string>();
+export function collectHashtagsFromInfos(itemInfos: ItemInfos) {
+  // {
+  //   '#pve': {
+  //     variants: {
+  //       '#PVE': 4,
+  //       '#pve': 2
+  //     },              <- hashtagCollection
+  //     count: 6        <- structure
+  //   }
+  // }
+  const hashtagCollection: NodeJS.Dict<{ variants: NodeJS.Dict<number>; count: number }> = {};
+
   for (const info of Object.values(itemInfos)) {
-    const matches = getHashtagsFromNote(info.notes);
-    if (matches) {
-      for (const match of matches) {
-        hashTags.add(match);
-      }
+    const hashtags = getHashtagsFromString(info.notes);
+    for (const h of hashtags) {
+      const lower = h.toLowerCase();
+      hashtagCollection[lower] ??= { count: 0, variants: {} };
+      hashtagCollection[lower].count++;
+      hashtagCollection[lower].variants[h] ??= 0;
+      hashtagCollection[lower].variants[h]++;
     }
   }
-  return uniqBy(hashTags, (t) => t.toLowerCase());
+
+  return Object.values(hashtagCollection)
+    .map((normalizedMeta) => {
+      const countsByVariant = Object.entries(normalizedMeta!.variants);
+      const mostPopularVariant = maxBy(countsByVariant, (v) => v[1]!)![0];
+      return [mostPopularVariant, normalizedMeta!.count] as [string, number];
```

Comment:
I think this can be:

```suggestion
      return [mostPopularVariant, normalizedMeta!.count] as const;
```

And it'll convince it to type the tuple correctly.